### ISSUE TITLE
Typed: Routers documentation example fails

### ISSUE THREAD

**Issue by PrettyCities** [akka/akka#27184]
The current Routers documentation includes the following snippet:
```scala
// #pool
// make sure the workers are restarted if they fail
val supervisedWorker = Behaviors.supervise(Worker.behavior).onFailure[Exception](SupervisorStrategy.restart)
val pool = Routers.pool(poolSize = 4)(supervisedWorker)
val router = ctx.spawn(pool, "worker-pool")

(0 to 10).foreach { n =>
  router ! Worker.DoLog(s"msg  a $n")
}
// #pool
```

However, when running the example, messages are not delivered due to the following error:
`[1] dead letters encountered. If this is not an expected behavior, then [Actor[akka://RouterSpec/user/$a/worker-pool#936913874]] may have terminated unexpectedly`

I'm new to Akka so I don't know too much about what's going on behind the scenes here. But, it appears that the `Routers.pool` actor is terminating before it can send messages to the `Worker` actors.

[Here is the source code that the Router docs make use of](https://github.com/akka/akka/blob/v2.5.23/akka-actor-typed-tests/src/test/scala/docs/akka/typed/RouterSpec.scala#L55-L57)

**Comment by johanandren** [comment]
Running `docs.akka.typed.RouterSpec` here without any dead letters, the workers are started on actor startup so I don't see how that can happen.

Are you sure it is not the group router you are seeing this with rather? It would pass the initial set of messages to dead letters before it saw the first listing from the receptionist, fixed in #26951

**Comment by PrettyCities** [comment]
Here is an image of the failing test ![Here is an image of the failing test](https://user-images.githubusercontent.com/15722891/59788831-32ff8d80-9292-11e9-9ff1-61dcbf528a83.png)

**Comment by johanandren** [comment]
Thanks for verifying. Curious!

**Comment by johanandren** [comment]
It is fixed in the 2.6 milestones but not because of #26951 but #26982

I think we need to backport that.

**Comment by johanandren** [comment]
Nope. I was wrong. The problem is that the test completes, and shuts down the actor system which stops the workers before all messages has been delivered. Turning it into a req-res just to be able to wait for all messages to be delivered before completing the test case would complicate the sample more than necessary though - this is mostly intended to show the API.

Perhaps just adding a note in the bottom of the test would be enough?

**Comment by PrettyCities** [comment]
Thank you for looking into this 😄

You mention that this was fixed in the 2.6 milestones. My project is using 2.5.23. When I copy and paste the router spec into a test file it fails to type check.

![image](https://user-images.githubusercontent.com/15722891/60965109-a23d3000-a2da-11e9-9f54-7cd1af5d19cb.png)
![image](https://user-images.githubusercontent.com/15722891/60965124-a8cba780-a2da-11e9-9801-46c29d7871a1.png)

Have there been API changes from 2.5-2.6 that are causing type checking to fail?

**Comment by johanandren** [comment]
Yeah, that is the change I mentioned earlier. Having the `Routers.pool` take a concrete behavior makes it easy to do the mistake of sharing a single instance of the behavior between the routers so we changed this into a factory in 2.6.

We have decided to collect all further API changes for typed into Akka 2.6, which we are currently working towards, rather than changing the APIs more in the 2.5 branch, so we likely won't backport that change.
